ldfparser/ldf.lark

```diff
@@ -1,124 +1,125 @@
-ldf_signal_default_value: C_INITIALIZER_LIST|C_INT 
-ldf_signal_size: C_INT 
-ldf_signal_bit_offset: C_INT 
-ldf_signal_name: CNAME 
-
-ldf_node_name: CNAME 
-
-ANY_OPENED_BLOCK: /.*{/
-ANY_CLOSED_BLOCK: /.*}/
+start: ldf
+
+ldf: (header_lin_description_file | header_protocol_version | header_language_version | header_speed | header_channel | nodes | node_compositions | signals | diagnostic_signals | frames | sporadic_frames | event_triggered_frames | diagnostic_frames | node_attributes | schedule_tables | signal_groups | signal_encoding_types | signal_representations)*
+
+ldf_identifier: CNAME
+ldf_integer: C_INT
+ldf_float: C_FLOAT
+
+// LIN 2.1 Specification, section 9.2.1
+header_lin_description_file: "LIN_description_file" ";"
+header_protocol_version: "LIN_protocol_version" "=" "\"" ldf_float "\"" ";"
+header_language_version: "LIN_language_version" "=" "\"" ldf_float "\"" ";"
+header_speed: "LIN_speed" "=" ldf_float "kbps" ";"
+header_channel: "Channel_name" "=" "\"" ldf_identifier "\"" ";"
+
+// LIN 2.1 Specification, section 9.2.2
+nodes: "Nodes" "{" nodes_master nodes_slaves "}"
+nodes_master: "Master:" ldf_identifier "," ldf_float "ms" "," ldf_float "ms" ";"
+nodes_slaves: "Slaves:" ldf_identifier ("," ldf_identifier)* ";"
+
+// LIN 2.1 Specification, section 9.2.2.3
+node_compositions: "composite" "{" (node_compositions_configuration+) "}"
+node_compositions_configuration: "configuration" ldf_identifier "{" (node_compositions_composite+) "}"
+node_compositions_composite: ldf_identifier "{" ldf_identifier ("," ldf_identifier)* "}"
+
+// LIN 2.1 Specification, section 9.2.3
+signals: "Signals" "{" (signal_definition)+ "}"
+signal_definition: ldf_identifier ":" ldf_integer "," signal_default_value "," ldf_identifier ("," ldf_identifier)+ ";"
+signal_default_value: signal_default_value_array | signal_default_value_single
+signal_default_value_single: C_INT
+signal_default_value_array: "{" C_INT ("," C_INT)* "}"
+
+// LIN 2.1 Specification, section 9.2.3.2
+diagnostic_signals: "Diagnostic_signals" "{" (ANY_SEMICOLON_TERMINATED_LINE)* "}"
+
+// LIN 2.1 Specification, section 9.2.4
+frames: "Frames" "{" [frame_definition]+ "}"
+frame_definition : ldf_identifier  ":" ldf_integer "," ldf_identifier ("," ldf_integer)? "{" frame_signals "}"
+frame_signals: (frame_signal+)
+frame_signal: ldf_identifier "," ldf_integer ";" 
+
+// LIN 2.1 Specification, section 9.2.4.2
+sporadic_frames: "Sporadic_frames" "{" (sporadic_frame_definition+) "}"
+sporadic_frame_definition: ldf_identifier ":" ldf_identifier ("," ldf_identifier)* ";"
+
+// LIN 2.1 Specification, section 9.2.4.3
+event_triggered_frames: "Event_triggered_frames" "{" (event_triggered_frame_definition+) "}"
+event_triggered_frame_definition: ldf_identifier ":" (ldf_identifier ",")? ldf_integer event_triggered_frame_definition_frames ";"
+event_triggered_frame_definition_frames: ("," ldf_identifier)+
+
+// LIN 2.1 Specification, section 9.2.4.4
+diagnostic_frames: "Diagnostic_frames" "{" "}"
```

Comment:
Missing content causes syntax error when the diagnostic frames are defined, should be something similar, but the content is actually specified in the LIN standard, section 9.2.4.4

```
diagnostic_frames: "Diagnostic_frames" "{" (diagnostic_frame_definition)+ "}"
diagnostic_frame_definition: ldf_identifier ":" ldf_integer "{" (ANY_SEMICOLON_TERMINATED_LINE)* "}"
```